snapcraft/internal/pluginhandler/_patchelf.py

```diff
@@ -193,6 +187,10 @@ def patch(self) -> None:
             dynamic_linker = elf.find_linker(
                 root_path=self._primedir, snap_base_path=self._snap_base_path
             )
+            logger.warning(
+                "libc6 has been staged into the snap: only do this if you know what "
+                "what you are doing."
```

Comment:
What are the consequences? How can they avoid staging libc?

Reply:
by using build environments and building there, which cannot be suggested yet.